satellite/console/service.go

```diff
@@ -121,6 +123,15 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 	if err != nil {
 		return nil, errs.New(internalErrMsg)
 	}
+	if user.PartnerID != "" {
+		offerType = rewards.Partner
+	}
+
+	//TODO: Create a current offer cache to replace database call
+	currentReward, err := s.rewards.GetCurrentByType(ctx, offerType)
+	if err != nil && err.Error() != rewards.NoCurrentOfferErr {
```

Comment:
This is checking whether `err` is a `rewards.NoCurrentOfferErr` by checking whether the stringified version of `err` is equal to the `rewards.NoCurrentOfferErr` string. This can be a fragile way to do things, because another developer might unknowingly change the error instance before it gets to this point (maybe they are adding helpful explanatory information about what's happening now, or they are `.Wrap()`ping the error in another error class, etc). If the error instance changes at all, this check will no longer work as intended.

But! There is a more idiomatic way to do this (well, idiomatic for our codebase, anyway) which also solves that problem. First, you would make `rewards.NoCurrentOfferErr` be an error class instead of a string. In satellite/rewards/rewards.go:

```go
var (
    MaxRedemptionErr  = errs.Class("This offer redemption has reached its capacity")
    NoCurrentOfferErr = errs.Class("no current offer")
)
```

Then, when returning one of those errors, you make a new err instance from the class. For example, in offers.go:

```go
    if err == sql.ErrNoRows {
        return nil, rewards.NoCurrentOfferErr.New("offerType=%d", offerType)
    }
```

(You don't have to use that format string; that's just an example of something that might be useful in a NoCurrentOfferErr instance. The empty string would work too.) Finally, when we want to check if the error is not from `NoCurrentOfferErr`, on this line of the code, you could do something like

```go
    if err != nil && !rewards.NoCurrentOfferErr.Has(err) {
```

That will still work if extra information is added to the err string or if the err instance gets `.Wrap()`ped in other errror classes too.